### ISSUE TITLE
tvOS detection bug

### ISSUE THREAD

**Issue by brad0** [mpv-player/mpv#7087]
Updating our OpenBSD port of mpv I came across what seems like odd behavior with the tvOS detection...

I don't have any such --enable-tvos option. I have used --disable--tvos as a workaround, but that seems odd on an OS that is not tvOS.

Setting top to                           : /home/ports/pobj/mpv-0.30.0/mpv-0.30.0
Setting out to                           : /home/ports/pobj/mpv-0.30.0/build-amd64
Checking for waf version in 1.8.4-2.1.0  : ok
Checking for program 'cc'                : cc
Checking for program 'pkg-config'        : /usr/bin/pkg-config
Checking for program 'ar'                : /usr/bin/ar
Checking for program 'rst2html'          : /usr/local/bin/rst2html
Checking for program 'rst2man'           : /usr/local/bin/rst2man
Checking for program 'rst2pdf'           : not found
Checking for program 'windres'           : not found
Checking for program 'perl'              : /usr/bin/perl
Checking for 'clang' (C compiler)        : cc
Detected target OS:                      : os-openbsd
Checking for compiler flags -std=c11     : yes
Checking for compiler flags -Werror -Werror=implicit-function-declaration : yes
Checking for compiler flags -Werror -Wno-error=deprecated-declarations    : yes
Checking for compiler flags -Werror -Wno-error=unused-function            : yes
Checking for compiler flags -Werror -Wempty-body                          : yes
Checking for compiler flags -Werror -Wdisabled-optimization               : yes
Checking for compiler flags -Werror -Wstrict-prototypes                   : yes
Checking for compiler flags -Werror -Wno-format-zero-length               : yes
Checking for compiler flags -Werror -Werror=format-security               : yes
Checking for compiler flags -Werror -Wno-redundant-decls                  : yes
Checking for compiler flags -Werror -Wvla                                 : yes
Checking for compiler flags -Werror -Wno-format-truncation                : no
Checking for LGPL (version 2.1 or later) build                            : disabled
Checking for GPL (version 2 or later) build                               : yes
Checking for internal audio filter chain                                  : yes
Checking for mpv CLI player                                               : yes
Checking for shared library                                               : yes
Checking for static library                                               : disabled
Checking for static build                                                 : disabled
Checking for whether to include binary compile time                       : yes
Checking for whether to optimize                                          : disabled
Checking for whether to compile-in debugging information                  : yes
Checking for manpage generation                                           : yes
Checking for html manual generation                                       : disabled
Checking for pdf manual generation                                        : disabled
Checking for dynamic loader                                               : yes
Checking for C plugins                                                    : yes
Checking for inline assembly (currently without effect)                   : yes
Checking for test suite (using cmocka)                                    : disabled
Checking for generate a clang compilation database                        : disabled
Checking for static Swift linking                                         : disabled
Checking for compiler support for noexecstack                             : yes
Checking for linker support for --nxcompat --no-seh --dynamicbase         : no
Checking for -lm                                                          : yes
Checking for MinGW                                                        : os-win32 not found
Checking for POSIX environment                                            : yes
Checking for Android environment                                          : no
Checking for tvOS environment                                             : no
You manually enabled the feature 'tvos', but the autodetection check failed.

**Comment by CounterPillow** [comment]
Please provide the `./build/config.log`.

**Comment by brad0** [comment]
[config.log](https://github.com/mpv-player/mpv/files/3774293/config.log)

**Comment by qmega** [comment]
Going to ping @eworm-de who maintains the official Arch package for mpv. I saw that `--disable-tvos` was added there, so perhaps this came up? (I'm *not* getting this issue on Arch Linux myself.)

@brad0 Another file that could be interesting is `.lock-waf_*`. I have `'enable_tvos': None` in there.

That config.log doesn't look suspicious to me, though of course I could be missing something.

**Comment by qmega** [comment]
Oh, nevermind, that config.log _is_ suspicious:
```
# waf 2.0.18 (abi 20, python 20710f0 on openbsd6)
```

mpv supports only waf 2.0.9.

**Comment by brad0** [comment]
That doesn't make any difference. Before I made an attempt at updating to 0.30.0 we were using waf 2.0.14 without issue. Trying with 2.0.9 doesn't change anything.

**Comment by qmega** [comment]
Okay, apparently waf is supposed to have compatibility within patch versions (2.0.n). I actually wasn't aware of that; it's notorious for its lack of compatibility between minor versions. You'll probably still get more sympathy here using the exact version, but sure, I guess that's not the problem in this case. If you've got 2.0.9 running, it's probably still better to use that for debugging, just to remove one more variable.

I'd still be interested to see your `.lock-waf_*` file.

**Comment by Akemi** [comment]
remove the `--enable-tv` flag from your configure. that feature was removed anyway. why the hell the not available `--enable-tv` enables `tvos` i have no idea, but it's probably a waf bug and hab those flags are handled.

**Comment by qmega** [comment]
Ah, waf does that thing where you can abbreviate options with any unambiguous prefix (e.g. `--enable-libar` will enable libarchive, but `--enable-liba` will complain that multiple options start with that). Probably intentional rather than a bug, but still annoying.

**Comment by brad0** [comment]
Oh sorry about that I missed removing --enable-tv. After removing that option that indeed was the issue. Somewhat odd behavior from WAF.

**Comment by CounterPillow** [comment]
In the future, if we remove configuration flags, we can enable them with noops that print a warning instead to avoid this behaviour I guess.

**Comment by brad0** [comment]
I removed some other flags like --enable-dvdread, --enable-zsh-comp, --disable-vaapi-glx, and --disable-vapoursynth-lazy as WAF threw an error indicating an invalid flag. Trying to have such behavior with as many other flags when they are removed would be a good idea.

**Comment by wm4** [comment]
Not sure why we should leave such flags, just because waf is being stupid? I'd rather try to stop waf from doing this.
On the other hand, --disable could be ignored if the feature doesn't exist.

**Comment by CounterPillow** [comment]
Apparently this abbreviated option idiocy is inherited by waf from Python's optparse, which for some reason they are still using. There is [no way to disable it](https://github.com/python/cpython/blob/3.8/Lib/optparse.py#L1480).

tldr python bad
